Userland/Libraries/LibJS/Runtime/ProxyObject.cpp

```diff
@@ -798,13 +798,24 @@ MarkedValueList ProxyObject::internal_own_property_keys() const
         return MarkedValueList { heap() };
 
     // 8. Let trapResult be ? CreateListFromArrayLike(trapResultArray, « String, Symbol »).
-    auto trap_result = create_list_from_array_like(global_object, trap_result_array, [](auto value) -> Result<void, ErrorType> {
-        if (!value.is_string() && !value.is_symbol())
-            return ErrorType::ProxyOwnPropertyKeysNotStringOrSymbol;
-        return {};
+    HashTable<StringOrSymbol> unique_keys;
+    auto trap_result = create_list_from_array_like(global_object, trap_result_array, [&global_object, &unique_keys](auto value) {
+        auto& vm = global_object.vm();
+        if (!value.is_string() && !value.is_symbol()) {
+            vm.throw_exception<TypeError>(global_object, ErrorType::ProxyOwnPropertyKeysNotStringOrSymbol);
+            return;
+        }
+        auto property_key = value.to_property_key(global_object);
+        if (vm.exception())
+            return;
+        unique_keys.set(property_key, AK::HashSetExistingEntryBehavior::Keep);
```

Comment:
I don't think that's correct, `to_property_key` is observable and can have side effects. I think we need to check for duplicate `Value`s instead.

Reply:
to_property_key can only have side effects if Type != String && type != Symbol, which is exactly what we check in the previous line.

Reply:
ah, right! The exception check threw me off, please change that to a `VERIFY(!vm.exception())` then :)

Reply:
oh right, i automatically added an exception check based on reflex, will do :)